internal/io/csv.go

```diff
@@ -98,6 +101,40 @@ func ReadCSV(reader io.Reader, conf CSVConfig) (map[string]interface{}, []string
 		}
 	}
 
+	if conf.MissingColumnNameAlias != "" {
+		for i, name := range headers {
+			if name == "" {
+				headers[i] = conf.MissingColumnNameAlias
+			}
+		}
+	}
+
+	if conf.AddColumnIndexToDuplicateColumns {
+		headersMap := make(map[string]int)
+		// Loop through all columsn and find the duplicates.
+		for i, h := range headers {
+			headersMap[h] = i
+		}
+		// For go through all if there is duplicate down the line add index
+		for i, h := range headers {
+			index, ok := headersMap[h]
+			if ok && i != index {
+				counter := 0
```

Comment:
I like this approach better than the previous since it's more straight forward to reason about. I would like to change the name of the config variable though since the new approach has nothing to do with the column indicies.
Perhaps `RenameDuplicateColumns`?